### PR TITLE
Deps for inline func

### PR DESCRIPTION
**Pull request by EFF**
#### Note 
No test were added for the zipper, tell me if you judge it necessary

### CODE REVIEW

**Comment by slvnperron** [comment]
@EFF Please let me know as soon as the CI build passes and I will review
Thanks

**Comment by EFF** [comment]
@slvnperron build fixed

File: lib/zipper.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+var fs      = require('fs'),
+    zip     = require('node-zip'),
+    path    = require('path');
+
+var DEPENDENCIES_DEFAULT_FOLDER = 'node_modules',
+    MAX_LAMBDA_ZIP_SIZE         = 25e+6;
```
Comment (slvnperron) [inline lib/zipper.js]: I like this indenting. Should we always do it this way?

File: lib/LambdaHelper.js [hunk 2]
```diff
@@ -219,15 +174,20 @@ var _createProxy = function (executionStore, queueInitializedPromise, promise) {
...
         var functionIdentifier = _getFunctionHash(func);
+        var handlerName = configs.name || 'default';
 
         if (!proxyStore.hasOwnProperty(functionIdentifier)) {
             var lambdaFunc = _lambdaize(func);
-            var uploadPromise = _uploadFunctionAsync(lambdaFunc, functionConfig, functionIdentifier);
+            var functionAsString = 'exports.' + handlerName + '=' + lambdaFunc + ';';
+
+            var zippedFunction  = zipper.zipFunction(functionAsString, handlerName, deps);
+                uploadPromise   = _uploadZipAsync(zippedFunction, configs, functionIdentifier, false)
```
Comment (slvnperron) [inline lib/LambdaHelper.js]: missing ```;```

File: lib/zipper.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    var node_modules_index = subFolders.indexOf('node_modules') + 1; // Zero-based indexes
+    var folders = subFolders.splice(0, node_modules_index + 1); // We select the next folder
+    return folders.join(path.sep); // TODO throw better errors if require fails
+};
+
+var _addDepsToZip = function(_zip, deps){
+    var depsFolders = deps.map(function(dep) {
+        return _getDependencyFolder(dep);
+    }).filter(function(dep){
+        return dep !== '.';
```
Comment (slvnperron) [inline lib/zipper.js]: Can you please add a comment here as a reminder of why we exclude those
thanks

**Comment by slvnperron** [comment]
Awesome job! So much cleaner. LGTM

**Pull request merged**